Neuron/Services/AppChain/AppChain+TransactionDetails.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  AppChain+TransactionDetails.swift
+//  Neuron
+//
+//  Created by 晨风 on 2018/11/16.
+//  Copyright © 2018 Cryptape. All rights reserved.
+//
+
+import Foundation
+import BigInt
+import Alamofire
+import AppChain
+import PromiseKit
+
+class AppChainTransactionDetails: TransactionDetails {
+    var content: String = ""
+    var gasUsed: BigUInt = 0
+    var quotaUsed: BigUInt = 0
+    var chainId: Int = 0
+    var chainName: String = ""
+    var errorMessage: String?
+
+    enum AppChainCodingKeys: String, CodingKey {
+        case content
+        case gasUsed
+        case quotaUsed
+        case chainId
+        case chainName
+        case errorMessage
+    }
+
+    override init() {
+        super.init()
+    }
+
+    required init(from decoder: Decoder) throws {
+        try super.init(from: decoder)
+        let values = try decoder.container(keyedBy: AppChainCodingKeys.self)
+        content = (try? values.decode(String.self, forKey: .content)) ?? ""
+        if let value = try? values.decode(String.self, forKey: .gasUsed) {
+            gasUsed = BigUInt(string: value) ?? 0
+        }
+        if let value = try? values.decode(String.self, forKey: .quotaUsed) {
+            quotaUsed = BigUInt(string: value) ?? 0
+        }
+        chainName = (try? values.decode(String.self, forKey: .chainName)) ?? ""
+        chainId = (try? values.decode(Int.self, forKey: .chainId)) ?? 0
+    }
+}
+
+private struct AppChainTransactionsResponse: Decodable {
+    let result: Result
+    struct Result: Decodable {
+        let count: UInt
+        let transactions: [AppChainTransactionDetails]
+    }
+}
+
+private struct AppChainTransactionResponse: Decodable {
+    let result: Result
+    struct Result: Decodable {
+        let transaction: AppChainTransactionDetails?
+    }
+}
+
+class AppChainErc20TransactionDetails: AppChainTransactionDetails {
+}
+
+extension AppChainNetwork {
+    func getTransactionHistory(walletAddress: String, page: UInt, pageSize: UInt) throws -> [AppChainTransactionDetails] {
+        let url = AppChainNetwork().host().appendingPathComponent("/api/transactions")
```

Comment:
Is `/api/transactions` public endpoints from AppChain CITA RPC node? If I recall correctly it's not available from all nodes. We might need to connect to ReBirth to get transaction list?